Kernel/Net/UDPSocket.cpp

```diff
@@ -63,8 +63,8 @@ int UDPSocket::protocol_receive(const KBuffer& packet_buffer, void* buffer, size
 
 int UDPSocket::protocol_send(const void* data, int data_length)
 {
-    auto adapter = adapter_for_route_to(peer_address());
-    if (!adapter)
+    auto routing_decision = route_to(peer_address(), local_address());
+    if (!routing_decision.adapter)
```

Comment:
Why are we not checking `next_hop` here?

Reply:
Probably because I forgot to :1st_place_medal: